### PR TITLE
update proguard configuration no to depend on Android's default configuration

### PR DESCRIPTION
**Pull request by zaki50**
related to #2927

This PR adds a ProGuard configuration for native methods.

In normal cases, users include android's default ProGuard configuration and it contains a configuration for native methods.
But including default configuration is not mandatory and users freely write their own configurations from scratch.

To allow such a usage, ProGuard configuration provided by Realm should not depend on default configuration.

### CODE REVIEW

**Comment by zaki50** [comment]
@realm/java please review this

**Comment by cmelchior** [comment]
Since we already have a handful of support cases for this, I like this idea 👍

**Comment by beeender** [comment]
:+1:  But this should go to `releases` branch

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Enhancements
 
 * Enhanced `Table.toString()` to show a PrimaryKey field details (#2903).
+* Updated ProGuard configuration in order not to depend on Andtoid's default configuraion(#2972).
```
Comment (cmelchior) [inline CHANGELOG.md]: Andtoids -> Android's

Comment (kneth) [inline CHANGELOG.md]: "configuration(" -> "configuration ("

**Comment by kneth** [comment]
:+1: with a minor comments.

File: CHANGELOG.md (hunk 1 above)
Comment (zaki50) [inline CHANGELOG.md]: fixed

**Comment by zaki50** [comment]
I'll recreate this PR against `releases` branch.

**Pull request closed without merging**